src/python/pants/base/exception_sink.py

```diff
@@ -212,6 +213,23 @@ def reset_log_location(cls, new_log_location: str) -> None:
         cls._pid_specific_error_fileobj = pid_specific_error_stream
         cls._shared_error_fileobj = shared_error_stream
 
+    @classmethod
+    def open_pid_specific_error_stream(cls, path):
+        try:
+            ret = safe_open(path, mode="w+")
+        except Exception:
+            raise
+
+        def unlink_if_empty():
+            if ret.read(1) == "":
```

Comment:
How do you know this file handle won't be closed by someone else between now and exit? How do you know they won't read from the file handle such that the offset is at the end such that the file is non, empty, but you read "" anyhow? Seems like this should seek to the beginning before the read for the latter potential issue and seems like you should re-open the file using the path in the closure to fix the 1st. That though fixes the second too.

Reply:
For #2 I beat your comment :P see `seek0` comment.

Reply:
Yup, but still broken by your same logic of being robust against a fh you hand out. Definitely want to re-open locally.

Reply:
I'm actually tossing the handle completely and going to use `os.path.getsize`

Reply:
Or just stat it and read the size.

Reply:
I think `getsize` is more readable considering they do the same thing :man_shrugging:

Reply:
Nay, [they are the same thing](https://github.com/python/cpython/blob/f4c03484da59049eb62a9bf7777b963e2267d187/Lib/genericpath.py#L50)